ydb/core/kqp/provider/yql_kikimr_exec.cpp

```diff
@@ -1628,22 +1636,69 @@ class TKiSinkCallableExecutionTransformer : public TAsyncCallbackTransformer<TKi
+                                    auto parseEnum = [] (const TString vectorType) {
+                                        if (vectorType == "float")
+                                            return Ydb::Table::VectorIndexSettings_VectorType_VECTOR_TYPE_FLOAT;
+                                        else if (vectorType == "uint8")
+                                            return Ydb::Table::VectorIndexSettings_VectorType_VECTOR_TYPE_UINT8;
+                                        else if (vectorType == "int8")
+                                            return Ydb::Table::VectorIndexSettings_VectorType_VECTOR_TYPE_INT8;
+                                        else if (vectorType == "bit")
+                                            return Ydb::Table::VectorIndexSettings_VectorType_VECTOR_TYPE_BIT;
+                                        else
+                                            YQL_ENSURE(false, "Wrong vector_type: " << vectorType);
+                                    };
+                                    protoVectorSettings.set_vector_type(parseEnum(vectorSetting.Value().Cast<TCoAtom>().StringValue()));
+                                } else if (vectorSetting.Name().Value() == "vector_dimension") {
+                                    auto parseInt = [] (const TString vectorDimensionStr) {
+                                        ui32 vectorDimension;
+                                        YQL_ENSURE(TryFromString(vectorDimensionStr, vectorDimension), "Wrong vector_dimension: " << vectorDimensionStr);
+                                        return vectorDimension;
+                                    };
+                                    protoVectorSettings.set_vector_dimension(parseInt(vectorSetting.Value().Cast<TCoAtom>().StringValue()));
+                                } else {
+                                    YQL_ENSURE(false, "Wrong vector setting name: " << vectorSetting.Name().Value());
+                                }
+                            }
+                        }
+                        else {
+                            ctx.AddError(TIssue(ctx.GetPosition(nameNode.Pos()), TStringBuilder() << "Unknown add vector index setting: " << name));
                             return SyncError();
                         }
                     }
-                    switch (add_index->type_case()) {
-                        case Ydb::Table::TableIndex::kGlobalIndex:
-                            *add_index->mutable_global_index() = Ydb::Table::GlobalIndex();
-                            break;
-                        case Ydb::Table::TableIndex::kGlobalAsyncIndex:
-                            *add_index->mutable_global_async_index() = Ydb::Table::GlobalAsyncIndex();
-                            break;
-                        default:
-                            YQL_ENSURE(false, "Unknown index type: " << (ui32)add_index->type_case());
-                    }
```

Comment:
Why was it removed?

Reply:
I have done it's intentionally. It's duplicate reaction on index type. All the necessary logic is upper `if (name == "indexType")`.